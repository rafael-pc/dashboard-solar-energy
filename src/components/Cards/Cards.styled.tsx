import styled from "styled-components";
import { theme } from "../../styles";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  margin: 0 auto;
`;

export const Content = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  text-align: center;
  width: 85%;
  gap: 30px;
  margin: 0 auto;
  margin-top: 30px;

  h3 {
    width: 100%;
    margin-bottom: 12px;
    padding: 8px;
    font-size: ${theme.font.sizes.xxxs};
    color: ${theme.colors.dark_light};
  }

  span {
    font-size: ${theme.font.sizes.l};
    color: ${theme.colors.dark_light};
  }

  @media only screen and (max-width: ${theme.screen.large_desktop}) {
    margin-top: 40px;
    gap: 15px;
  }

  @media only screen and (max-width: ${theme.screen.desktop}) {
    margin-top: 60px;
    gap: 20px;
  }

  @media only screen and (max-width: ${theme.screen.mobile}) {
    margin-top: 20px;
    gap: 10px;
  }

  @media only screen and (max-height: ${theme.screen.mobile}) {
    margin-top: 15px;
    gap: 5px;
  }
`;

export const Card = styled.div`
  display: flex;
  flex-basis: 195px;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: 140px;
  padding: 10px;
  border-radius: ${theme.border.radius.m};
  font-weight: ${theme.font.bold};
  color: ${theme.colors.dark};
  border: 1px solid ${theme.colors.neutral_400};
  background-color: ${theme.colors.neutral_100};

  div {
    width: 100%;
  }

  &:hover {
    color: ${theme.colors.bg_primary};
    border: 1px solid ${theme.colors.bg_primary};

    h3 {
      color: ${theme.colors.bg_primary};
    }

    span {
      color: ${theme.colors.bg_primary};
    }
  }

  @media only screen and (max-width: ${theme.screen.large_desktop}) {
    flex-basis: 150px;
    height: 130px;
    padding: 5px;

    h3 {
      font-size: ${theme.font.sizes.xxxs};
      margin-bottom: 8px;
      padding: 7px 0;
    }

    span {
      font-size: ${theme.font.sizes.m};
    }
  }

  @media only screen and (max-width: ${theme.screen.desktop}) {
    flex-basis: 150px;
    height: 150px;

    h3 {
      font-size: ${theme.font.sizes.xxxs};
      padding: 7px 0;
    }

    span {
      font-size: ${theme.font.sizes.m};
    }
  }

  @media only screen and (max-width: ${theme.screen.mobile}) {
    flex-basis: 147px;
    height: 147px;
    padding: 5px;

    h3 {
      font-size: ${theme.font.sizes.xxxs};
      margin-bottom: 15px;
      padding: 7px 0;
    }

    span {
      font-size: ${theme.font.sizes.m};
    }
  }

  @media only screen and (max-height: ${theme.screen.mobile}) {
    flex-basis: 105px;
    height: 105px;
    padding: 5px;

    h3 {
      font-size: ${theme.font.sizes.xxxxxs};
      margin-bottom: 2px;
      padding: 7px 0;
    }

    span {
      font-size: ${theme.font.sizes.s};
    }
  }
`;
