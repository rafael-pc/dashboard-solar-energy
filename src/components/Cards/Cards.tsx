import { FC } from "react";
import * as S from "./Cards.styled";

interface IData1Item {
  status: string;
}

interface IData2Item {
  energia: number;
}

interface ICardsProps {
  data1: IData1Item[];
  data2: IData2Item[];
}

const Cards: FC<ICardsProps> = ({ data1, data2 }) => {
  const total = data2.reduce((acc, valor) => acc + valor.energia, 0) / data2.length;
  const ativo = data1.filter((unit) => unit.status === "ativo");
  const inativo = data1.filter((unit) => unit.status === "inativo");

  return (
    <S.Container>
      <S.Content>
        <S.Card>
          <div>
            <h3>Total unidades</h3>
            <span>{data1.length}</span>
          </div>
        </S.Card>
        <S.Card>
          <div>
            <h3>Unidades ativas</h3>
            <span>{ativo.length}</span>
          </div>
        </S.Card>
        <S.Card>
          <div>
            <h3>Unidades inativas</h3>
            <span>{inativo.length}</span>
          </div>
        </S.Card>
        <S.Card>
          <div>
            <h3>MÃ©dia de energia</h3>
            <span>{total.toFixed(0) + "W"}</span>
          </div>
        </S.Card>
      </S.Content>
    </S.Container>
  );
};

export default Cards;
