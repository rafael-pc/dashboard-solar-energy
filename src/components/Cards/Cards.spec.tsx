import { render } from "@testing-library/react";
import Cards from "./Cards";

describe("Cards", () => {
  it("renders correctly", () => {
    const data1 = [
      { status: "ativo" },
      { status: "ativo" },
      { status: "inativo" },
    ];
    const data2 = [
      { energia: 10 },
      { energia: 20 },
      { energia: 30 },
    ];
    const { container } = render(<Cards data1={data1} data2={data2} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
