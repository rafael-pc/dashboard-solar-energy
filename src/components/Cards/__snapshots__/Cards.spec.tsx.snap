// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards renders correctly 1`] = `
<div
  class="sc-bcPKhP kUlJtQ"
>
  <div
    class="sc-grXZZQ gKirki"
  >
    <div
      class="sc-dkjKgF eWCqCD"
    >
      <div>
        <h3>
          Total unidades
        </h3>
        <span>
          3
        </span>
      </div>
    </div>
    <div
      class="sc-dkjKgF eWCqCD"
    >
      <div>
        <h3>
          Unidades ativas
        </h3>
        <span>
          2
        </span>
      </div>
    </div>
    <div
      class="sc-dkjKgF eWCqCD"
    >
      <div>
        <h3>
          Unidades inativas
        </h3>
        <span>
          1
        </span>
      </div>
    </div>
    <div
      class="sc-dkjKgF eWCqCD"
    >
      <div>
        <h3>
          MÃ©dia de energia
        </h3>
        <span>
          20W
        </span>
      </div>
    </div>
  </div>
</div>
`;
